backend/server/src/nix/mod.rs

```diff
@@ -0,0 +1,50 @@
+use anyhow::Result;
+use std::process::Command;
+use tracing::{debug, info};
+use std::collections::HashMap;
+
+/// Given a drv, traverse all direct drv dependencies
+pub fn traverse_drvs (drv_path: &str) -> Result<HashMap<String, Vec<String>>> {
+    debug!("traversing {}", drv_path);
+
```

Comment:
we should probably do an initial check to see if this drv is in the drv cache to avoid having to allocate a hashmap and beginning the traversal at all

Reply:
Agreed